Core/clim-basic/clipboard.lisp

```diff
@@ -0,0 +1,76 @@
+(in-package :climi)
+
+;;;
+;;;  Functions dealing with copying to the clipboard.
+;;;
+
+(deftype representation-type-name () '(member :string :html :image))
+
+(defgeneric copy-to-clipboard-with-port (port sheet clipboard-p object presentation-type)
+  (:documentation "Method to be implemented by backends."))
+
+(defmethod copy-to-clipboard-with-port ((port clim:port) (sheet clim:sheet) clipboard-p object presentation-type)
+  "Fallback implementation if the implementation does not support a clipboard."
+  nil)
+
+(defun copy-to-clipboard (sheet object &key presentation-type)
+  "Copy OBJECT to the clipboard.
+SHEET is the owner of the clipboard, and it is not guaranteed that the
+content of the clipboard will be available after the sheet has been
+removed."
+  (copy-to-clipboard-with-port (port sheet) sheet t object presentation-type))
+
+(defun copy-to-selection (sheet object &key presentation-type)
+  "Copy OBJECT to the selection.
+SHEET is the owner of the selection, and it is not guaranteed that the
+content of the selection will be available after the sheet has been
+removed."
+  (copy-to-clipboard-with-port (port sheet) sheet nil object presentation-type))
+
+(defgeneric clear-clipboard-with-port (port sheet clipboard-p))
+
+(defmethod clear-clipboard-with-port (port sheet clipboard-p)
+  "Fallback implementation if the implementation does not support a clipboard.."
+  nil)
+
+(defun clear-clipboard (sheet)
+  (clear-clipboard-with-port (port sheet) sheet t))
+
+(defun clear-selection (sheet)
+  (clear-clipboard-with-port (port sheet) sheet nil))
```

Comment:
same story, I don't understand this interface duplication.

Reply:
This is similar to `COPY-TO-SELECTION` and `COPY-TO-CLIPBOARD` where the two user-facing functions are directed to a single backend implementation. Just like in that case, the term *clipboard* in the backend function might not be ideal, since it actually works with both the selection and the clipboard.

Reply:
I've added some documentation to this function.